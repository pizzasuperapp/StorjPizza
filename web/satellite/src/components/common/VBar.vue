// Copyright (C) 2019 Storj Labs, Inc.
// See LICENSE for copying information.

<template>
    <div class="bar-container">
        <div class="bar-container__fill" :style="barFillStyle" />
    </div>
</template>

<script lang="ts">
import { Component, Prop, Vue } from 'vue-property-decorator';

/**
 * BarFillStyle class holds info for BarFillStyle entity.
 */
class BarFillStyle {
    'background-color': string;
    width: string;

    public constructor(backgroundColor: string, width: string) {
        this['background-color'] = backgroundColor;
        this.width = width;
    }
}

// @vue/component
@Component
export default class VBar extends Vue {
    @Prop({default: 0})
    private readonly current: number;
    @Prop({default: 0})
    private readonly max: number;
    @Prop({default: '#0068DC'})
    private readonly color: string;

    public get barFillStyle(): BarFillStyle {
        if (this.current > this.max) {
            return new BarFillStyle(this.color, '100%');
        }

        const width = (this.current / this.max) * 100 + '%';

        return new BarFillStyle(this.color, width);
    }
}
</script>

<style scoped lang="scss">
    .bar-container {
        width: 100%;
        height: 13px;
        margin-top: 10px;
        border-radius: 4px;
        background-color: #f4f6f9;
        position: relative;

        &__fill {
            max-width: 100%;
            height: 100%;
            position: absolute;
            left: 0;
            top: 0;
            border-radius: 20px;
        }
    }
</style>

