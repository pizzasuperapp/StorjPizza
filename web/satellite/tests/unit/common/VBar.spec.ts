// Copyright (C) 2020 Storj Labs, Inc.
// See LICENSE for copying information.

import VBar from '@/components/common/VBar.vue';

import { mount } from '@vue/test-utils';

describe('VBar.vue', () => {
    it('renders correctly', () => {
        const wrapper = mount(VBar, {
            propsData: {
                current: 500,
                max: 1000,
            },
        });

        expect(wrapper).toMatchSnapshot();
    });

    it('renders correctly if current > max', () => {
        const wrapper = mount(VBar, {
            propsData: {
                current: 1000,
                max: 500,
            },
        });

        expect(wrapper).toMatchSnapshot();

        const el = wrapper.find('.bar-container__fill').element as HTMLElement;
        expect(el.style.width).toMatch('100%');
    });
});
