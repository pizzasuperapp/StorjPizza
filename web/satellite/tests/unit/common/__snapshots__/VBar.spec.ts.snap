// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VBar.vue renders correctly 1`] = `
<div class="bar-container">
  <div class="bar-container__fill" style="background-color: rgb(0, 104, 220); width: 50%;"></div>
</div>
`;

exports[`VBar.vue renders correctly if current > max 1`] = `
<div class="bar-container">
  <div class="bar-container__fill" style="background-color: rgb(0, 104, 220); width: 100%;"></div>
</div>
`;
